### ISSUE TITLE
Base.quadgk fails with Unitful

### ISSUE THREAD

**Issue by ggggggggg** [ajkeller34/Unitful.jl#44]
My guess is that this issue actually arises from Base, but it seems worth having someone with more knowledge look at it first. The [docs for `quadgk`](http://docs.julialang.org/en/release-0.5/stdlib/math/#Base.quadgk) claim that it should work with any type that implements, `+`, `-`, multiplication by real values, and `norm`; all of which seem to work with `Quantity`.

```julia
julia> using Unitful

julia> quadgk(x->x*1u"mm",0,1)
ERROR: Unitful.DimensionError()
 in convfact(...) at /Users/oneilg/.julia/v0.5/Unitful/src/Conversion.jl:79
 in convert at /Users/oneilg/.julia/v0.5/Unitful/src/Conversion.jl:183 [inlined]
 in Base.QuadGK.Segment(::Float64, ::Float64, ::Unitful.Quantity{Float64,Unitful.Dimensions{(Unitful.Dimension{:Length}(1//1),)},Unitful.Units{(Unitful.Unit{:Meter}(-3,1//1),),Unitful.Dimensions{(Unitful.Dimension{:Length}(1//1),)}}}, ::Unitful.Quantity{Float64,Unitful.Dimensions{(Unitful.Dimension{:Length}(1//1),)},Unitful.Units{(Unitful.Unit{:Meter}(-3,1//1),),Unitful.Dimensions{(Unitful.Dimension{:Length}(1//1),)}}}) at ./quadgk.jl:40
 in evalrule(::##1#2, ::Float64, ::Float64, ::Array{Float64,1}, ::Array{Float64,1}, ::Array{Float64,1}, ::Base.LinAlg.#vecnorm) at ./quadgk.jl:79
 in do_quadgk(::##1#2, ::Array{Float64,1}, ::Int64, ::Type{Float64}, ::Float64, ::Float64, ::Int64, ::Base.LinAlg.#vecnorm) at ./quadgk.jl:120
 in #quadgk#15(::Array{Any,1}, ::Function, ::Function, ::Int64, ::Int64) at ./quadgk.jl:230
 in quadgk(::Function, ::Int64, ::Int64) at ./quadgk.jl:225
```

**Comment by ajkeller34** [comment]
I have a fix for this but it won't be ready until julia 0.6. When it is ready, then you should be able to integrate Quantity-valued functions and also integrate over a unitful domain.

**Comment by ajkeller34** [comment]
Actually, the fix is ready now on Julia 0.5. Just clone [UnitfulIntegration.jl](https://github.com/ajkeller34/UnitfulIntegration.jl) and it should work with the most recent release of Unitful and the [QuadGK.jl](https://github.com/JuliaMath/QuadGK.jl) package. I will tag and release UnitfulIntegration sometime this month. The functionality is located in a separate package to avoid adding dependencies to Unitful. 

Be sure to do `QuadGK.quadgk(...)` instead of just `quadgk` on Julia 0.5 otherwise you'll call the `quadgk` implementation in Julia Base which is not compatible. In Julia 0.6 quadgk is no longer in Base so this won't be a problem.
